lib/foreman_tasks/tasks/export_tasks.rake

```diff
@@ -0,0 +1,244 @@
+        case step.state
+        when :success
+          "success"
+        when :error
+          "important"
+        end
+      end
+
+      def progress_width(step)
+        if step.state == :error
+          100 # we want to show the red bar in full width
+        else
+          step.progress_done * 100
+        end
+      end
+
+      def step(step_id)
+        @plan.steps[step_id]
+      end
+
+
+      def updated_url(new_params)
+        url("?" + Rack::Utils.build_nested_query(params.merge(new_params.stringify_keys)))
+      end
+
+
+    end
+
+    class PageHelper
+      def self.pagify(template)
+       pre =  <<-HTML
+       <html>
+       <head>
+       <title>Dynflow Console</title>
+       <script src="jquery.js"></script>
+       <link rel="stylesheet" type="text/css" href="bootstrap.css">
+       <link rel="stylesheet" type="text/css" href="application.css">
+       <script src="bootstrap.js"></script>
+       <script src="run_prettify.js"></script>
+       <script src="application.js"></script>
+       </head>
+       <body>
+          #{template}
+       <body>
+       </html>
+         HTML
+       end
+
+       def self.copy_assets(tmp_dir)
+           ['vendor/bootstrap/js/bootstrap.js',
+            'vendor/google-code-prettify/run_prettify.js',
+            'vendor/jquery/jquery.js',
+            'vendor/jquery/jquery.js',
+            'javascripts/application.js',
+            'vendor/bootstrap/css/bootstrap.css',
+            'stylesheets/application.css'].each do |file|
+
+             filename = File.join(Gem::Specification.find_by_name("dynflow").gem_dir, 'web', 'assets', file)
+             FileUtils.copy_file(filename, File.join(tmp_dir, File.basename(file)))
+           end
+       end
+
+      def self.generate_index(tasks)
+        html = "<div><table class=\"table\">"
+        tasks.order("started_at desc").all.each do |task|
+          html << "<tr><td><a href=\"#{task.id}.html\">#{task.label}</a></td><td>#{task.started_at}</td>\
+                   <td>#{task.state}</td><td>#{task.result}</td></tr>"
+        end
+        html << "</table></div>"
+      end
+    end
+
+
+    if ENV['tasks'] == 'all'
+      tasks = ForemanTasks::Task
+    else
+      tasks = ForemanTasks::Task.where("result != 'success'")
+    end
+
+    export_filename = ENV['export'] || "/tmp/task-export-#{DateTime.now.to_i}.tar.gz"
+
+
+    Dir.mktmpdir('task-export') do |tmp_dir|
+      PageHelper.copy_assets(tmp_dir)
+
+
+      renderer = TaskRender.new
+      count = 1
+      total = tasks.count
+
+      tasks.all.each do |task|
+        File.open(File.join(tmp_dir, "#{task.id}.html"), 'w') {|file| file.write(PageHelper.pagify(renderer.render_task(task)))}
+        puts "#{count}/#{total}"
+        count += 1
+      end
+
+      File.open(File.join(tmp_dir, "index.html"), 'w') {|file| file.write(PageHelper.pagify(PageHelper.generate_index(tasks)))}
+
+      system("tar cvzf #{export_filename} #{tmp_dir} > /dev/null")
```

Comment:
`s/system/sh/g` http://rake.rubyforge.org/classes/FileUtils.html#M000018 the behaviour is the same as system, but the important benefit of usnig it is, if something fails when calling the command, the user gets an error describing the problem and the rake task fails without continuing.